client/src/components/scheduler/RescheduleCancelAppointment.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import {
+  Typography,
+  Grid,
+  makeStyles,
+  TextField,
+  Button,
+} from "@material-ui/core";
+import { Link } from "react-router-dom";
+import axios from "axios";
+
+const RescheduleCancelAppointment = ({ variant, appointmentId, eventUrl }) => {
+  const reschedule = variant === "reschedule";
+  const cancel = variant === "cancel";
+  const classes = useStyles();
+  const cancelAppointment = (event) => {
+    axios.delete(`/api/appointment/cancel/${appointmentId}`).then(() => {
+      setAppointmentCancelled(true);
+    });
+  };
+
+  const linkToReschedule = `/appt/${eventUrl}`;
+  const linkToSuccessfulCancellation = "/appointment/cancelled";
+  return (
+    <div className={classes.gridContainer}>
+      <Grid container direction="column" spacing={reschedule ? 1 : 4}>
+        <Grid item>
+          <Typography variant="h5">
+            {reschedule && "Cancel appointment and reschedule?"}
+            {cancel && "Cancel appointment?"}
+          </Typography>
+        </Grid>
+        <Grid item>
+          {cancel && (
+            <TextField
+              label="Reason for canceling"
+              multiline
+              rows={7}
+              variant="outlined"
+              fullWidth
+            />
+          )}
+        </Grid>
+        <Grid item>
+          <Button className={classes.button} onClick={cancelAppointment}>
+            <Link
```

Comment:
I wonder if linking them to reschedule at the same time as sending the request means that the availability shown will assume this meeting is still happening. I also wonder if that matters since they're rescheduling anyway

Reply:
Right that's something I haven't considered. This downside sounds acceptable to me though since we are choosing to simplify the rescheduling flow. Can definitely discuss this during stand up!

Reply:
update: This is actually resolved after I let the parent component handle redirection.